nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java

```diff
@@ -234,4 +361,10 @@ public void process(final OutputStream out) throws IOException {
             }
         }
     }
+
+    @OnPrimaryNodeStateChange
+    public void onPrimaryNodeChange(final PrimaryNodeState newState) {
+        isPrimaryNode = (newState == PrimaryNodeState.ELECTED_PRIMARY_NODE);
```

Comment:
(adding for transparency) @ijokarumawak and I talked out of band. Current thinking is to add another variable to the Initialization Contexts in order to know when the component is created if it is clustered or not and if it is on the primary node.